### PR TITLE
gui-libs/wlroots: various fixes; add myself as maintainer

### PR DESCRIPTION
**Pull request by sevz17**

### CODE REVIEW

File: gui-libs/wlroots/wlroots-9999.ebuild
Comment (arthurzam) [inline gui-libs/wlroots/wlroots-9999.ebuild:1]: I feel like this file is very out of sync with 0.16.2-r2 you add in this PR. Can you sync the ebuilds?

**Review by arthurzam — CHANGES_REQUESTED** [review]

File: gui-libs/wlroots/metadata.xml [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
 	</maintainer>
+	<maintainer type="person" proxied="yes">
+		<email>leohdz172@proton.me</email>
+		<name>Leonardo Hernández Hernández</name>
+		<description>Additionally the upstream maintainer</description>
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
```
[the remark this answers is not in the document]
Comment (sevz17) [reply gui-libs/wlroots/metadata.xml]: Ohh, I thought proxy maintainers project would be needed whenever a non-gentoo maintainer is listed, good to know.

I also deleted my description, I am not a upstream maintainer, it was a copy-paste from another package, sorry.

File: gui-libs/wlroots/wlroots-9999.ebuild
Comment (sevz17) [reply gui-libs/wlroots/wlroots-9999.ebuild:1]: Yeah, but it's because it has new USE flags, I really think I can't sync it.

**Comment by sevz17** [comment]
All done

File: gui-libs/wlroots/metadata.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sevz17) [reply gui-libs/wlroots/metadata.xml]: Given https://github.com/gentoo/gentoo/pull/33084#discussion_r1338145491, what do I do?

File: gui-libs/wlroots/wlroots-0.16.2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	SLOT="0/$(ver_cut 2)"
+fi
+
+LICENSE="MIT"
+IUSE="+drm +libinput tinywl vulkan x11-backend xcb-errors X"
+REQUIRED_USE="
+	xcb-errors? ( || ( x11-backend X ) )
+"
+
+DEPEND="
```
Comment (thesamesam) [inline gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:27]: fwiw we usually sort all unconditional deps first (alphabetically), and then all conditional deps (by USE flag)

File: gui-libs/wlroots/wlroots-0.16.2-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	sys-auth/seatd:=
+	virtual/libudev
+	vulkan? (
+		dev-util/glslang:0=
+		dev-util/vulkan-headers:0=
+		media-libs/vulkan-loader:0=
+	)
+	>=x11-libs/libdrm-2.4.114:0=
+	x11-libs/libxkbcommon
+	>=x11-libs/pixman-0.42.0:0=
```
Comment (thesamesam) [inline gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:43]: If the only slot is :0, just write :=.

File: gui-libs/wlroots/wlroots-0.16.2-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		x11-libs/xcb-util-wm
+	)
+	xcb-errors? ( x11-libs/xcb-util-errors )
+"
+RDEPEND="
+	${DEPEND}
+"
+BDEPEND="
+	>=dev-libs/wayland-protocols-1.28
+	>=dev-util/meson-0.60.0
```
Comment (thesamesam) [inline gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:60]: meson.eclass already does `>=dev-util/meson-0.62.2` so you can drop this

File: gui-libs/wlroots/wlroots-0.16.2-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	virtual/pkgconfig
+"
+
+src_configure() {
+	local backends=(
+		$(usev drm)
+		$(usev libinput)
+		$(usev x11-backend 'x11')
+	)
+	# Separate values with a comma with this evil floating point bit hack
```
Comment (thesamesam) [inline gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:71]: why is it talking about floating point?

Comment (sevz17) [reply gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:71]: I don't know either, do I drop the comment?

Also, do I have to add the proxy-maintainers project, or as arthurzam said it's unnecessary?.

Comment (thesamesam) [reply gui-libs/wlroots/wlroots-0.16.2-r2.ebuild:71]: yes, drop it, it makes no sense. As for proxy-maint: it depends on if arthur or prom are going to handle merging changes from you, or if the proxied maint project will be hepling. So, if he says it's unnecessary, no need. But we're happy to have you too.

**Review by arthurzam — APPROVED** [review]

**Pull request closed without merging**